contrib/devtools/fix-copyright-headers.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+current_year=$(date +%Y)
+
+git grep "// Copyright (c) .* The Zcash developers" \
+  | awk -F ':' '{print $1}' \
+  | xargs -I {} sed -i"" -e "s#// Copyright (c) \([0-9]\{4\}\)\(-[0-9]\{4\}\)\? The Zcash developers#// Copyright (c) \1-$current_year The Zcash developers#" {}
+
+git commit -a -F - << EOS
+scripted-diff: Update Zcash copyrights to $current_year
+
+-BEGIN VERIFY SCRIPT-
+git grep "// Copyright (c) .* The Zcash developers" \
+  | awk -F ':' '{print $1}' \
+  | xargs -I {} sed -i"" -e "s#// Copyright (c) \([0-9]\{4\}\)\(-[0-9]\{4\}\)\? The Zcash developers#// Copyright (c) \1-$current_year The Zcash developers#" {}
```

Comment:
For some reason, this fails the scripted diff lint (which apparently isn't able to reproduce any of the date updates).

Separately, how easy would it be to make this a pseudo-quine? That is, the script that is placed between the `-BEGIN VERIFY SCRIPT-` and `-END VERIFY SCRIPT-` is precisely the script that was used?

Reply:
I made some attempts at this, but the escaping is just a bear.

Reply:
Aha, got it.